### PR TITLE
Implement duplications filter and quick disconnect filter for isOnline notifications

### PR DESCRIPTION
**Pull request by alxmiron**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -18,23 +18,50 @@ const withOnlineStatus = WrappedComponent =>
...
         </>
       );
     }
   };
 
 class App extends React.Component {
-  componentDidUpdate({ isOnline }) {
-    NotificationManager.info(isOnline ? 'Online' : 'Offline');
+  componentDidUpdate() {
+    NotificationManager.info(this.props.isOnline ? 'Online' : 'Offline');
```
Comment (alxmiron) [inline src/App.js]: Notification text should match status text under the buttons, as I understand

File: src/App.js [hunk 2]
```diff
@@ -18,23 +18,50 @@ const withOnlineStatus = WrappedComponent =>
...
+        return; // Skip, because it's already online
+      }
+      if (nextIsOnline && !isOnline) {
+        this.setState({ isOnline: true }); // Switch to online, skipping duplications
+      } else if (!nextIsOnline && isOnline) {
+        if (this.offlineTimer) clearTimeout(this.offlineTimer); // Clear previous deferred offline switch, if exists
+        this.offlineTimer = setTimeout(this.setOffline, this.quickDisconnectMS); // Set new deferred switch to offline
+        this.lastTimeOffline = now;
+      }
+    }
```
Comment (alxmiron) [inline src/App.js]: I decided to use a pure JS solution.  
The logic looks transparent to me

**Comment by alxmiron** [comment]
I suppose I should remove my fork after your review because it's visible for other candidates, who receive this task

File: src/App.js (hunk 1 above)
Comment (grebenyuksv-preply) [reply src/App.js]: good catch 👍

File: src/App.js (hunk 2 above)
Comment (grebenyuksv-preply) [reply src/App.js]: Looks great to me. Indeed, very bulletproof and clear 👍 Just one small question:
**Input**:
`Online — Offline, Offline, Offline, Offline, Offline, ...`
**Output Notifications**:
Online. It only prints `Offline` 2 seconds after the _last_ `Offline` event of the series. 
**The output that sounds more natural to me**
Online, Offline, and then silence. I mean, I'd expect `Offline` to appear 2 seconds after the _first_ `Offline` event of the series.

Was that an understatement from us or your conscious decision?

Comment (alxmiron) [reply src/App.js]: Sorry, I missed that during testing. 
Fixed now, and made it even more simple

**Comment by grebenyuksv-preply** [comment]
@alxmiron yeah, we wouldn't mind that, although we haven't seen any solutions which would resemble yours. Could you please remove it from public visibility?

**Pull request closed without merging**